openrtb_ext/request.go

```diff
@@ -314,3 +317,203 @@ func (m ExtMultiBid) String() string {
 	}
 	return fmt.Sprintf("{Bidder:%s, Bidders:%v, MaxBids:%s, TargetBidderCodePrefix:%s}", m.Bidder, m.Bidders, maxBid, m.TargetBidderCodePrefix)
 }
+
+func (erp *ExtRequestPrebid) Clone() *ExtRequestPrebid {
+	if erp == nil {
+		return nil
+	}
+
+	clone := *erp
+	clone.Aliases = maputil.Clone(erp.Aliases)
+	clone.AliasGVLIDs = maputil.Clone(erp.AliasGVLIDs)
+	clone.BidAdjustmentFactors = maputil.Clone(erp.BidAdjustmentFactors)
+
+	if erp.BidderConfigs != nil {
+		clone.BidderConfigs = make([]BidderConfig, len(erp.BidderConfigs))
+		for i, bc := range erp.BidderConfigs {
+			clonedBidderConfig := BidderConfig{Bidders: sliceutil.Clone(bc.Bidders)}
+			if bc.Config != nil {
+				config := &Config{}
+				if bc.Config.ORTB2 != nil {
+					config.ORTB2 = &ORTB2{
+						Site: maputil.Clone(bc.Config.ORTB2.Site),
+						App:  maputil.Clone(bc.Config.ORTB2.App),
+						User: maputil.Clone(bc.Config.ORTB2.User),
+					}
+				}
+				clonedBidderConfig.Config = config
+			}
+			clone.BidderConfigs[i] = clonedBidderConfig
+		}
+	}
+
+	if erp.Cache != nil {
+		clone.Cache = &ExtRequestPrebidCache{}
+		if erp.Cache.Bids != nil {
+			clone.Cache.Bids = &ExtRequestPrebidCacheBids{}
+			if erp.Cache.Bids.ReturnCreative != nil {
+				clone.Cache.Bids.ReturnCreative = ptrutil.ToPtr(*erp.Cache.Bids.ReturnCreative)
```

Comment:
Consider using `ptrutil.Clone` here to save on the nil check, like you do in some of the other methods.